worlds/dark_souls_3/__init__.py

```diff
@@ -1375,106 +1376,125 @@ def post_fill(self):
+                locations.sort()
+                spheres_per_player[player].append(locations)
+
+        for ds3_world in ds3_worlds:
+            locations_by_sphere = spheres_per_player[ds3_world.player]
+
+            # All items in the base game in approximately the order they appear
+            all_item_order: List[DS3ItemData] = [
+                item_dictionary[location.default_item_name]
+                for region in region_order
+                # Shuffle locations within each region.
+                for location in ds3_world._shuffle(location_tables[region])
+                if ds3_world._is_location_available(location)
             ]
 
-            # It's expected that there may be more total items than there are matching locations if
-            # the player has chosen a more limited accessibility option, since the matching
-            # locations *only* include items in the spheres of accessibility.
-            if len(converted_item_order) < len(all_matching_locations):
-                raise Exception(
-                    f"DS3 bug: there are {len(all_matching_locations)} locations that can " +
-                    f"contain smoothed items, but only {len(converted_item_order)} items to smooth."
-                )
-
-            for sphere in locations_by_sphere:
-                locations = [loc for loc in sphere if loc.item.name in names]
-
-                # Check the game, not the player, because we know how to sort within regions for DS3
-                offworld = self._shuffle([loc for loc in locations if loc.game != "Dark Souls III"])
-                onworld = sorted((loc for loc in locations if loc.game == "Dark Souls III"),
-                                 key=lambda loc: loc.data.region_value)
-
-                # Give offworld regions the last (best) items within a given sphere
-                for location in onworld + offworld:
-                    new_item = self._pop_item(location, converted_item_order)
-                    location.item = new_item
-                    new_item.location = location
-
-        if self.options.smooth_upgrade_items:
-            base_names = {
-                "Titanite Shard", "Large Titanite Shard", "Titanite Chunk", "Titanite Slab",
-                "Titanite Scale", "Twinkling Titanite", "Farron Coal", "Sage's Coal", "Giant's Coal",
-                "Profaned Coal"
-            }
-            smooth_items([item for item in all_item_order if item.base_name in base_names])
-
-        if self.options.smooth_soul_items:
-            smooth_items([
-                item for item in all_item_order
-                if item.souls and item.classification != ItemClassification.progression
-            ])
+            # All DarkSouls3Items for this world that have been assigned anywhere, grouped by name
+            full_items_by_name: Dict[str, List[DarkSouls3Item]] = defaultdict(list)
+            for location in ds3_world.multiworld.get_filled_locations():
```

Comment:
```suggestion
            for location in multiworld.get_filled_locations():
```

Similar change below